src/PeopleSDKAdapter.js

```diff
@@ -0,0 +1,120 @@
+import {from, fromEvent, concat} from 'rxjs';
+import {deconstructHydraId} from '@webex/common';
+import {PeopleAdapter, PersonStatus} from '@webex/component-adapter-interfaces';
+import {flatMap, filter, finalize, map, publishReplay, refCount} from 'rxjs/operators';
+
+const USER_PRESENCE_UPDATE_EVENT = 'event:apheleia.subscription_update';
+
+/**
+ * The `PeopleSDKAdapter` is an implementation of the `PeopleAdapter` interface.
+ * This adapter utilizes the Webex JS SDK to fetch data about a person.
+ *
+ * @class PeopleSDKAdapter
+ * @extends {PeopleAdapter}
+ */
+export default class PeopleSDKAdapter extends PeopleAdapter {
+  constructor(datasource) {
+    super(datasource);
+
+    this.getPersonObservables = {};
+  }
+
+  /**
+   * Returns a proper Enum key for a given PersonStatus value
+   *
+   * @private
+   * @param {string} status
+   * @returns {string} PersonStatus
+   * @memberof PeopleSDKAdapter
+   */
+  getStatus(status) {
+    return Object.keys(PersonStatus).find((key) => PersonStatus[key] === status);
+  }
+
+  /**
+   * Returns the UUID of a person from the given public ID
+   *
+   * @private
+   * @param {string} ID
+   * @returns {string}
+   * @memberof PeopleSDKAdapter
+   */
+  getPersonUUID(ID) {
+    return deconstructHydraId(ID).id; // ciscospark://us/PEOPLE/<UUID>
+  }
+
+  /**
+   * Fetches the person data from the sdk and returns in the shape required by adapter.
+   *
+   * @private
+   * @param {string} ID
+   * @returns {Person}
+   * @memberof PeopleSDKAdapter
+   */
+  async fetchPerson(ID) {
+    const {id, emails, displayName, firstName, lastName, avatar, orgId} = await this.datasource.people.get(ID);
+
+    return {
+      ID: id,
+      emails,
+      displayName,
+      firstName,
+      lastName,
+      avatar,
+      orgID: orgId,
+    };
+  }
+
+  /**
+   * Returns an observable that emits person data of the given ID.
+   *
+   * @public
+   * @param {string} ID ID of person to get.
+   * @returns {Observable.<Person>}
+   * @memberof PeopleSDKAdapter
+   */
+  getPerson(ID) {
+    if (!(ID in this.getPersonObservables)) {
+      const person$ = from(this.fetchPerson(ID));
+
+      // Subscribe to `Apheleia` internal service to listen for presence changes
+      // And update the person object with status response from the subscription
+      const presenceSubscription$ = from(this.datasource.internal.presence.subscribe(this.getPersonUUID(ID))).pipe(
+        map((data) => data.responses[0].status.status)
+      );
+      const personStatus$ = presenceSubscription$.pipe(
+        flatMap((status) => person$.pipe(map((person) => ({...person, status: this.getStatus(status)}))))
```

Comment:
Do we have to fetch the person data (`person$`) for every update or can we store the result of the initial fetch and just modify it with the changed status?

Reply:
we are not fetching. `from` stores the data for you automatically. Unless you invoke the method/from operator again.